### PR TITLE
Test docker build image

### PR DESCRIPTION
**Pull request by mshiyk**

### CODE REVIEW

File: .circleci/old_config.yml [hunk 1]
```diff
@@ -0,0 +1,824 @@
+version: 2.1
```
Comment (dsueltenfuss) [inline .circleci/old_config.yml:1]: Is there a reason you have an old copy of the config checked in?

File: BuilderImage.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#FROM cimg/node:16.16.0
```
Comment (dsueltenfuss) [inline BuilderImage.Dockerfile:1]: This file needs to be cleaned up. All of the additional commented out commands are unnecessary.

File: .circleci/config.yml [hunk 3]
```diff
@@ -27,7 +27,6 @@ commands:
           command: |
             sudo apt update
             sudo apt install -y openjdk-11-jdk-headless
-      - install-awscli
```
Comment (dsueltenfuss) [inline .circleci/config.yml:30]: If this step is no longer being used, the step should be removed as well

File: .circleci/config.yml [hunk 4]
```diff
@@ -390,13 +385,40 @@ commands:
...
+            docker push ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>-latest
+
+  docker-deploy-image:
+    steps:
+      - configure
+      - setup_remote_docker:
+          version: 20.10.14
+          docker_layer_caching: true
+      - build-and-push-docker-image:
+          image-name: test
```
Comment (dsueltenfuss) [inline .circleci/config.yml:411]: This name should be more descriptive of what the image is. Naming an image test won't tell the user what it is used for,

File: .circleci/config.yml [hunk 5]
```diff
@@ -390,13 +385,40 @@ commands:
             git status
             gh pr create --fill --base main --head $CIRCLE_BRANCH
 
+  build-and-push-docker-image:
```
Comment (dsueltenfuss) [inline .circleci/config.yml:388]: This step name should be more descriptive. What docker image are we building and pushing?

File: .circleci/config.yml [hunk 6]
```diff
@@ -540,6 +583,23 @@ jobs:
...
 
+  #Maybe have a move this to a image folder
+  docker-deploy:
+    docker:
+      - image: cimg/node:16.16.0
+    steps:
+      - set-dev-env-vars
+      - docker-deploy-image
+
+  ecr-deploy:
```
Comment (dsueltenfuss) [inline .circleci/config.yml:594]: I don't see this job being used anywhere. If its not being used, it should be removed

File: .circleci/config.yml [hunk 7]
```diff
@@ -540,6 +583,23 @@ jobs:
           enable-dashboard: << parameters.enable-dashboard >>
           suite: << parameters.suite >>
 
+  #Maybe have a move this to a image folder
+  docker-deploy:
```
Comment (dsueltenfuss) [inline .circleci/config.yml:587]: This job name should properly describe what the job does. The job doesn't deploy anything to docker, it builds the builder image. Deploying to docker makes me think you are launching a container

File: .circleci/config.yml [hunk 8]
```diff
@@ -540,6 +583,23 @@ jobs:
...
+    docker:
+      - image: cimg/node:16.16.0
+    steps:
+      - set-dev-env-vars
```
Comment (dsueltenfuss) [inline .circleci/config.yml:591]: Why are you setting the dev envs in this step?

File: .circleci/config.yml [hunk 9]
```diff
@@ -27,7 +27,6 @@ commands:
           command: |
             sudo apt update
             sudo apt install -y openjdk-11-jdk-headless
```
Comment (dsueltenfuss) [inline .circleci/config.yml:29]: This command should be moved to the docker image, and removed from the CircleCI workflow

**Review by dsueltenfuss — CHANGES_REQUESTED** [review]

File: BuilderImage.Dockerfile (hunk 2 above)
Comment (mshiyk) [reply BuilderImage.Dockerfile:1]: Cleaned out commented out commands.

File: .circleci/config.yml (hunk 3 above)
Comment (mshiyk) [reply .circleci/config.yml:30]: Removing that apt install step.

File: .circleci/config.yml (hunk 4 above)
Comment (mshiyk) [reply .circleci/config.yml:411]: Renamed the image to node_v16_16

Comment (dsueltenfuss) [reply .circleci/config.yml:411]: Thats still not a good name for this image. The image name should describe what the image is for. In this case, its a builder for the NJ BFS Navigator project

File: .circleci/config.yml (hunk 5 above)
Comment (mshiyk) [reply .circleci/config.yml:388]: Renamed the job to this.
build-and-push-custom-node-image

File: .circleci/config.yml (hunk 6 above)
Comment (mshiyk) [reply .circleci/config.yml:594]: Removed the extra job.

File: .circleci/config.yml (hunk 7 above)
Comment (mshiyk) [reply .circleci/config.yml:587]: Changed the name.

File: .circleci/config.yml (hunk 9 above)
Comment (mshiyk) [reply .circleci/config.yml:29]: Moved it to the docker image.

**Review by dsueltenfuss — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 8 above)
Comment (mshiyk) [reply .circleci/config.yml:591]: In order to mitigate the following error.

https://app.circleci.com/pipelines/github/newjersey/navigator.business.nj.gov/10630/workflows/f08e5d91-6386-4aae-bb4d-f3cf812e2111/jobs/47431

File: .circleci/config.yml [hunk 10]
```diff
@@ -502,18 +541,18 @@ jobs:
...
-      - image: cimg/node:16.16.0-browsers
+      - image: $AWS_ACCOUNT_ID_DEV.dkr.ecr.$AWS_REGION.amazonaws.com/navigator_builder:node_v16_16
+        aws_auth:
+          aws_access_key_id: $AWS_ACCESS_KEY_ID_DEV
+          aws_secret_access_key: $AWS_SECRET_ACCESS_KEY_DEV
     steps:
       - set-cypress-env-vars
-      - attach_workspace:
-          at: .
+      - persist_to_workspace:
```
Comment (dsueltenfuss) [inline .circleci/config.yml:550]: Why was this changed from attach_workspace to persist_to_workspace? 
As this is the integration test, you want to be using the workspace that has been saved previously. There wouldn't be anything to persist at this point

File: .circleci/config.yml [hunk 11]
```diff
@@ -390,13 +380,41 @@ commands:
...
+
+  docker-setup-node-image:
+    steps:
+      - install-awscli
+      - setup_remote_docker:
+          version: 20.10.14
+          docker_layer_caching: true
+      - checkout
+      - build-and-push-custom-node-image:
+          image-name: node_v16_16
```
Comment (dsueltenfuss) [inline .circleci/config.yml:407]: This image name is still confusing. While this image is based on node, its not what the image is. The image is a builder for the BFS navigator.

File: .circleci/config.yml [hunk 12]
```diff
@@ -390,13 +380,41 @@ commands:
...
+          command: docker build -f BuilderImage.Dockerfile . -t ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>-${CIRCLE_SHA1} -t ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >> -t ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>-latest
+      - run:
+          name: Push Builder Docker Image
+          command: |
+            aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com
+            docker push ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>-${CIRCLE_SHA1}
+            docker push ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>
+            docker push ${AWS_ACCOUNT_ID_DEV}.dkr.ecr.${AWS_REGION}.amazonaws.com/navigator_builder:<< parameters.image-name >>-latest
+
+  docker-setup-node-image:
```
Comment (dsueltenfuss) [inline .circleci/config.yml:399]: This step name is confusing. Why is it referencing node?

**Review by dsueltenfuss — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 10 above)
Comment (mshiyk) [reply .circleci/config.yml:550]: Reverted it back to attach_workspace.

File: .circleci/config.yml (hunk 11 above)
Comment (mshiyk) [reply .circleci/config.yml:407]: Changed the name to bfs_navigator_builder.

File: .circleci/config.yml (hunk 12 above)
Comment (mshiyk) [reply .circleci/config.yml:399]: Changed it to docker-setup-navigator-builder-image.
